lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,83 @@
+import { getManagerConfig, mergeChildConfig } from '../../../config';
+import type {
+  RegexManagerTemplates,
+  RenovateConfig,
+} from '../../../config/types';
+import { getManagerList } from '../../../modules/manager';
+import { validMatchFields } from '../../../modules/manager/regex/utils';
+import type { CustomExtractConfig } from '../../../modules/manager/types';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig {
+  managerList: Set<string>;
+  managers: WorkerExtractConfig[];
+}
+
+function getRegexManagerFields(
+  config: WorkerExtractConfig
+): CustomExtractConfig {
+  const regexFields = {} as CustomExtractConfig;
+  for (const field of validMatchFields.map(
+    (f) => `${f}Template` as keyof RegexManagerTemplates
+  )) {
+    if (config[field]) {
+      regexFields[field] = config[field];
+    }
+  }
+
+  return {
+    autoReplaceStringTemplate: config.autoReplaceStringTemplate,
+    matchStrings: config.matchStrings,
+    matchStringsStrategy: config.matchStringsStrategy,
+    ...regexFields,
+  };
+}
+
+function getFilteredManagerConfig(
+  config: WorkerExtractConfig
+): WorkerExtractConfig {
+  return {
+    ...(config.manager === 'regex' ? getRegexManagerFields(config) : {}),
+    manager: config.manager,
+    fileMatch: config.fileMatch,
+    npmrc: config.npmrc,
+    npmrcMerge: config.npmrcMerge,
+    enabled: config.enabled,
+    ignorePaths: config.ignorePaths ?? [],
+    includePaths: config.includePaths ?? [],
+    skipInstalls: config.skipInstalls,
+    registryAliases: config.registryAliases,
+    fileList: [],
+  };
+}
+
+export function generateFingerprintConfig(
+  config: RenovateConfig
+): FingerprintExtractConfig {
+  let managerExtractConfigs: WorkerExtractConfig[] = [];
+  let managerList = new Set(getManagerList());
+  const { enabledManagers } = config;
+  if (enabledManagers?.length) {
+    managerList = new Set(enabledManagers);
+  }
+
+  for (const manager of managerList) {
+    const managerConfig = getManagerConfig(config, manager);
+    if (manager === 'regex') {
+      for (const regexManager of config.regexManagers ?? []) {
+        managerExtractConfigs.push({
+          ...mergeChildConfig(managerConfig, regexManager),
+          fileList: [],
```

Comment:
why do we enforce a empty file list? 🤔

Reply:
If we would have gone normally then a fileList would be generated and added to manager config after cloning.
But since we are trying to skip that part we need to add rmpty fileList to satisfy the TS